openml/datasets/dataset.py

```diff
@@ -456,6 +456,10 @@ def _create_pickle_in_cache(self, data_file: str) -> Tuple[str, str, str]:
                     # The file is likely corrupt, see #780.
                     # We deal with this when loading the data in `_load_data`.
                     return data_pickle_file, data_feather_file, feather_attribute_file
+                except Exception:
```

Comment:
Could you specifically catch `ModuleNotFoundError` and any others that are expected? I would prefer not to use a catch-all as if new issues arise, we could possibly just fix them instead.

Reply:
You can add a check for the error of #898 too. And document in comments in one sentence why an error is expected. Including the issue number is good, but since they are easily summarized it is nice to be able to have the documentation within the code, e.g.
```
catch ModuleNotFoundError:
    # 780: Pickled dataframe is likely of pandas<1.0 while attempting to load with pandas>=1.0
    return ...
catch ValueError:
   (maybe check for the specific message)
   # 898: Pickled dataframe pickled with protocol 5 (Py3.8), but loaded with protocol 4 (<Py3.8).
   return ...
```

Reply:
Good point, I just updated the code.